holoviews/core/dimension.py

```diff
@@ -344,13 +344,15 @@ def relabel(self, label=None, group=None, depth=0):
         Assign a new label and/or group to an existing LabelledData
         object, creating a clone of the object with the new settings.
         """
-        keywords = [('label',label), ('group',group)]
-        obj = self.clone(self.data,
-                         **{k:v for k,v in keywords if v is not None})
+        new_data = self.data
         if (depth > 0) and getattr(obj, '_deep_indexable', False):
-            for k, v in obj.items():
-                obj[k] =  v.relabel(group=group, label=label, depth=depth-1)
```

Comment:
Reimplemented this because assigning this way is inefficient for an internal operation.